<a href="https://colab.research.google.com/github/NairMpc/DataScience/blob/main/Assignment_6_ci.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Background

In quality control processes, especially when dealing with high-value items, destructive sampling is a necessary but costly method to ensure product quality. The test to determine whether an item meets the quality standards destroys the item, leading to the requirement of small sample sizes due to cost constraints.

Scenario

A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.

Data

A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:

1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

Assignment Tasks

a. Build 99% Confidence Interval Using Sample Standard Deviation

Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.

b. Build 99% Confidence Interval Using Known Population Standard Deviation

If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.



In [ ]:
# given data
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

### a. Build 99% Confidence Interval Using Sample Standard Deviation

In [ ]:
import numpy as np
import scipy


In [ ]:
#converting to numpy array
data = np.array(data)
data

array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.2 ,
       1.33, 1.18, 1.22, 1.29])

In [ ]:
# Required parameters
n = len(data)
df = n-1
ci = 0.99
alpha = 1 - ci
alpha/2

0.0050000000000000044

a. Build 99% Confidence Interval Using Sample Standard Deviation

Using Sample standard deviation it is t_distribution

Confidence_interval = x bar +/- margin_of_error

margin_of_error = t_critical * s/sqrt of n

t_critica from tables : using df, alpha/2 = .005, ci = 0.99

also

t_critical = stats.t.ppf((1+ci)/2,df) or stats.t.ppf((1-alpha/2),df)

(scipy.stats.t.ppf())

In [ ]:
mean = np.mean(data)
mean

np.float64(1.2386666666666666)

In [ ]:
s = np.std(data)
s

np.float64(0.18661427836285438)

In [ ]:
t_critical = scipy.stats.t.ppf((1+ci)/2,df)
t_critical

np.float64(2.976842734370834)

In [ ]:
margin_error = t_critical * s/np.sqrt(n)
margin_error

np.float64(0.14343499804054374)

In [ ]:
lower_t = mean - margin_error
upper_t = mean + margin_error
print(f"Confidence Interval :({lower_t:.4f}, {upper_t:.4f})")

Confidence Interval :(1.0952, 1.3821)


Based on the results, we can conclude with 99% confidence that the true mean durability of the print-heads, in millions of characters, lies between 1.0952 and 1.3821. This means if we were to repeat this sampling and interval estimation process many times, 99% of the confidence intervals constructed would contain the true population mean durability.



###b. Build 99% Confidence Interval Using Known Population Standard Deviation

given that    

sigma = 0.2 million characters

As population standard deviation is known we use z_distribution

Confidence interval = mean (+/-) z_critical * sigma / sqrt of n

In [ ]:
z_critical = scipy.stats.norm.ppf((1+ci)/2)
z_critical

np.float64(2.5758293035489004)

In [ ]:
sigma = 0.2
lower_z = mean - z_critical*sigma/np.sqrt(n)
upper_z = mean + z_critical*sigma/np.sqrt(n)
print(f" Confidence Interval :({lower_z:.4f}, {upper_z:.4f})")


 Confidence Interval :(1.1057, 1.3717)


### Conclusion for 99% Confidence Interval Using Known Population Standard Deviation

When the population standard deviation is known (σ = 0.2 million characters), we can state with 99% confidence that the true mean durability of the print-heads (in millions of characters) lies within the interval of (1.1057, 1.3717). This interval, derived using the z-distribution, suggests a slightly narrower range compared to the t-distribution result, which is expected when there is less uncertainty due to a known population standard deviation.